src/zcbor_decode.c

```diff
@@ -42,10 +42,22 @@ do {\
 	} \
 } while(0)
 
-#define CHECK_MAJOR_TYPE(exp_type) \
-do { \
-	if (MAJOR_TYPE(*state->payload) != exp_type) { \
-		ZCBOR_ERR(ZCBOR_ERR_WRONG_TYPE); \
+static bool initial_checks(zcbor_state_t *state, zcbor_major_type_t exp_major_type)
```

Comment:
"Hopefull" - could we check that the binary gets smaller?

Reply:
Thanks, I was able to make some changes to make it smaller. Now, for the mcuboot boot_serial use case, the change makes no difference in size (the new function adds as much as is saved in three other functions), so if a program uses more than those three functions they should see a reduction in code size.